### PR TITLE
Remove `logical` argument of `vec_as_subscript2()`

### LINKED ISSUE #1608: `vec_as_subscript2()` has a misleading error when given logical input with `logical = "cast"`
Look at the first message:

> Subscript has the wrong type `logical`.

> It must be logical, numeric, or character.

``` r
library(vctrs)

vec_as_subscript2(TRUE, logical = "cast")
#> Error:
#> ! Must extract element with a single valid subscript.
#> ✖ Subscript has the wrong type `logical`.
#> ℹ It must be logical, numeric, or character.

vec_as_subscript2(TRUE, logical = "error")
#> Error:
#> ! Must extract element with a single valid subscript.
#> ✖ Subscript has the wrong type `logical`.
#> ℹ It must be numeric or character.
```

Also, why do we even expose a `logical` argument? Shouldn't it always be error? We internally error right before exiting if we have a logical argument.

### PR DESCRIPTION
**Pull request by lionel-**
Closes #1608

### CODE REVIEW

**Comment by lionel-** [comment]
Will need to update tidyselect and tibble. And keep the argument around in vctrs for some while in case users update vctrs but not tidyselect or tibble.

File: R/subscript.R [hunk 1]
```diff
@@ -64,36 +64,42 @@ vec_as_subscript_result <- function(i,
...
                               call = caller_env()) {
-  check_dots_empty0(...)
+  check_dots <- function(..., logical = "error") {
+    if (!is_string(logical, "error")) {
+      abort(
+        "`vctrs::vec_as_subscript2(logical = 'cast')` is deprecated.",
+        call = caller_env()
+      )
+    }
+    check_dots_empty0(...)
```
Comment (DavisVaughan) [inline R/subscript.R:78]: Do we want to tweak the `call` of `check_dots_empty0()` to also be `caller_env()`?

File: R/subscript.R [hunk 2]
```diff
@@ -126,7 +131,6 @@ vec_as_subscript2_result <- function(i,
   if (typeof(i) == "logical") {
```
Comment (DavisVaughan) [inline R/subscript.R:131]: Can you remove this section now? That's what I was hoping for

**Review by DavisVaughan — APPROVED** [review]

File: R/subscript.R (hunk 1 above)
Comment (lionel-) [reply R/subscript.R:78]: yup good catch

File: R/subscript.R (hunk 2 above)
Comment (lionel-) [reply R/subscript.R]: Thanks, now simplified.

**Pull request merged**